.github/workflows/inspektor-gadget.yml

```diff
@@ -225,8 +228,27 @@ jobs:
             IMAGE_TAG="latest"
         fi
         echo IMAGE_TAG=$IMAGE_TAG >> $GITHUB_ENV
-    - name: Build gadget default container
+    - name: Cross build gadget container for amd64 and arm64
       uses: docker/build-push-action@v2
+      # We cross build only on main because it takes a lot of time.
+      if: github.ref == 'main'
+      with:
+        context: /home/runner/work/inspektor-gadget/inspektor-gadget/
+        file: /home/runner/work/inspektor-gadget/inspektor-gadget/gadget-default.Dockerfile
+        build-args: |
+          ENABLE_BTFGEN=true
+        # TODO: how to avoid pushing a container before running integration tests
+        # Answer: push to runner registry first, if integration tests are OK
+        # push to final registry
+        push: true
+        tags: ${{ secrets.CONTAINER_REPO }}:${{ env.IMAGE_TAG }}
+        cache-from: type=local,src=/tmp/.buildx-cache
+        cache-to: type=local,dest=/tmp/.buildx-cache-new
+        # For the moment, we only support these two platforms.
+        platforms: linux/amd64,linux/arm64
```

Comment:
This step is a copy/paste and only the `platforms` field changed. Is it possible to do that instead?

```
${{ fromJSON('["linux/amd64", "linux/amd64,linux/arm64"]')[github.ref == 'main'] }}
```

(idea from [here](https://github.community/t/do-expressions-support-ternary-operators-to-change-their-returned-value/18114/4))

Reply:
It seems a good idea! Let's try this!